### PR TITLE
HPCC-24647 Improve record translation handling/warnings

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@jakesmith Please review

File: rtl/eclrtl/rtldynfield.cpp [hunk 1]
```diff
@@ -34,14 +34,22 @@
...
     else if (strieq(val, "alwaysDisk") || strieq(val, "disk"))
+    {
+        if (!isLocal)
+            throw makeStringException(0, "alwaysDisk translation mode can only be set via HINT");
         return RecordTranslationMode::AlwaysDisk;
+    }
     else if (strieq(val, "alwaysECL") || strieq(val, "ecl"))
+    {
+        if (!isLocal)
+            throw makeStringException(0, "alwaysDisk translation mode can only be set via HINT");
```
Comment (jakesmith) [inline rtl/eclrtl/rtldynfield.cpp:50]: should say "alwaysECL ..."

File: ecl/hthor/hthor.cpp [hunk 2]
```diff
@@ -8133,7 +8133,7 @@ CHThorDiskReadBaseActivity::CHThorDiskReadBaseActivity(IAgentContext &_agent, un
     {
         const char *recordTranslationModeHintText = _node->queryProp("hint[@name='layouttranslation']/@value");
         if (recordTranslationModeHintText)
-            recordTranslationModeHint = getTranslationMode(recordTranslationModeHintText);
+            recordTranslationModeHint = getTranslationMode(recordTranslationModeHintText, true);
```
Comment (jakesmith) [inline ecl/hthor/hthor.cpp:8136]: not directly related to this PR, but looks like hthor diskread only supports hint and not workunit option?
(and also IndexRead, Fetch and KJ)

**Comment by jakesmith** [comment]
@richardkchapman - looks good, just 1 issue with exception text

File: ecl/hthor/hthor.cpp (hunk 2 above)
Comment (richardkchapman) [reply ecl/hthor/hthor.cpp:8136]: I will open a Jira to investigate

**Comment by richardkchapman** [comment]
@ghalliday Please review/merge

File: rtl/eclrtl/rtldynfield.cpp (hunk 1 above)
Comment (richardkchapman) [reply rtl/eclrtl/rtldynfield.cpp]: Fixed

**Comment by ghalliday** [comment]
@richardkchapman the changes look good, but it looks like a few regression tests need to be updated to reflect the new restriction.

**Comment by richardkchapman** [comment]
@ghalliday Tests updated.

**Pull request merged**